### ISSUE TITLE
vcpkg crashes on missing license post-build lint warning

### ISSUE THREAD

**Issue by anders-wind** [microsoft/vcpkg#28845]
**Describe the bug**
Fail to build private package without copyright/license on the new 2023.01.09 release

**Environment**
- OS: Ubuntu 22.04 
- Compiler: gcc-12 and clang-15

**To Reproduce**
1. have a private package from a private registry, without a copyright/license
2. Upgrade to the new release (previous release 2022.11.14 works). (Only need to have the checkout newer, the builtin-baseline does not affect the result)
3. Run cmake configure step, and use the VCPKG_TOOLCHAIN to consume the library mentioned above.

**Expected behavior**
To have the package build.

**Failure logs**
For our coverage build we use GCC:
```
Preset CMake variables:

  CMAKE_BUILD_TYPE="Debug"
  CMAKE_CUDA_ARCHITECTURES="all-major"
  CMAKE_CXX_COMPILER_LAUNCHER="ccache"
  CMAKE_CXX_FLAGS="-std=c++20 -ffold-simple-inlines -pedantic -pedantic-errors -Wall -Wextra -Wpedantic -Wunreachable-code -Wuninitialized -Wcast-qual -Wshadow -Wformat=2 -Wundef -Wdeprecated -Wdouble-promotion -Wold-style-cast -Wno-ctad-maybe-unsupported -fno-common -Wno-unused-local-typedefs -Wno-unknown-pragmas -Winvalid-pch"
  CMAKE_CXX_FLAGS_DEBUG="-g -Og --coverage"
  CMAKE_CXX_STANDARD="20"
  CMAKE_C_COMPILER_LAUNCHER="ccache"
  CMAKE_EXE_LINKER_FLAGS_DEBUG="--coverage"
  CMAKE_EXPORT_COMPILE_COMMANDS:BOOL="TRUE"
  CMAKE_INSTALL_PREFIX:PATH="/opt/actions-runner/_work/our_consuming_lib/our_consuming_lib/bin"
  CMAKE_MAP_IMPORTED_CONFIG_DEBUG="RelWithDebInfo;Release;Debug;"
  CMAKE_SHARED_LINKER_FLAGS_DEBUG="--coverage"
  CMAKE_TOOLCHAIN_FILE="/opt/actions-runner/_work/our_consuming_lib/our_consuming_lib/vcpkg/scripts/buildsystems/vcpkg.cmake"
  CMAKE_UNITY_BUILD:BOOL="TRUE"
  CMAKE_UNITY_BUILD_BATCH_SIZE="16"
  THREADS_PREFER_PTHREAD_FLAG:BOOL="TRUE"
  LINKER_VARIABLE="lld"

Preset environment variables:

  CC="gcc-12"
  CUDAHOSTCXX="gcc-11"
  CUDA_PATH="/usr/local/cuda-11.7"
  CXX="g++-12"
  VCPKG_ROOT="/opt/actions-runner/_work/our_consuming_lib/our_consuming_lib/vcpkg"

-- Running vcpkg install
Fetching registry information from https://github.com/our_company/vcpkg-registry (main)...
Detecting compiler hash for triplet x64-linux...
The following packages will be built and installed:

...

Installing 130/131 our_lib:x64-linux...
Building our_lib[core]:x64-linux...
-- Installing port from location: /home/runner/.cache/vcpkg/registries/git-trees/b0c2acb052eadf59044b4618921b798b663a2e85
-- Using cached /opt/actions-runner/_work/our_consuming_lib/our_consuming_lib/vcpkg/downloads/our_lib-aacef56df5ae3706481bc5bb8c4fe880bd8f0ee7.tar.gz
-- Cleaning sources at /opt/actions-runner/_work/our_consuming_lib/our_consuming_lib/vcpkg/buildtrees/our_lib/src/80bd8f0ee7-a93f7f24c0.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /opt/actions-runner/_work/our_consuming_lib/our_consuming_lib/vcpkg/downloads/our_lib-aacef56df5ae3706481bc5bb8c4fe880bd8f0ee7.tar.gz
-- Using source at /opt/actions-runner/_work/our_consuming_lib/our_consuming_lib/vcpkg/buildtrees/our_lib/src/80bd8f0ee7-a93f7f24c0.clean
-- Configuring x64-linux
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Performing post-build validation
INTERNAL ERROR: failed to format default format string for index 649
format string: The software license must be available at ${CURRENT_PACKAGES_DIR}/share/{spec}/copyright
-- Running vcpkg install - failed
CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:883 (message):
  vcpkg install failed.  See logs for more information:
  /opt/actions-runner/_work/our_consuming_lib/our_consuming_lib/build/vcpkg-manifest-install.log
Call Stack (most recent call first):
  /usr/local/share/cmake-3.25/Modules/CMakeDetermineSystem.cmake:124 (include)
  CMakeLists.txt:8 (project)
```
(I have changed the names of our consuming library and the library we consume)

**Additional context**
My guess is that its failing at injecting `{spec}` here.
I was told to tag @vicroms :)

**Comment by vicroms** [comment]
Easiest repro is to try to install a package without a license. The localized string contains an unescaped sequence that vcpkg fails to format properly when trying to print the warning.

**Comment by dg0yt** [comment]
@vicroms Shouldn't this issue stay open until a new tool release is maded (available to boostrap)?

**Comment by vicroms** [comment]
Yes, it should. It auto-closed due to being referenced on the tool PR.

**Comment by senstar-nross** [comment]
@vicroms @FrankXie05 What is the expected ETA for a new tool release?  I just ran into this and am now in a bit of a build mess... Forced to move forward because of #28896 but now this fails...

**Comment by vicroms** [comment]
@senstar-nross 

I'd say soon. But in the meantime, you could modify the port to install a dummy copyright file as a workaround.

**Comment by senstar-nross** [comment]
@vicroms Thanks, it was not my port but then I went and looked I found that the copyright issue was fixed in the latest port.  The baseline in vcpkg is also updated to that port version. 

 I thought if the baseline was updated then I would get that version.  I guess I am still a bit confused as to how baselines work.

The port in question is quirc. I do not depend on it directly it is a dependency of opencv. OpenCV does not specify a version>= or override AFAIK.  When I build with the lastest vcpkg with the baseline hash set to the latest commit it chooses to build quirc 1.1#2 not 1.1#3 (which is the version specified in the baseline). If I add quirc 1.1#3 to my vcpkg.json it uses that version.
